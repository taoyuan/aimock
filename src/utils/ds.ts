import * as fs from 'fs';
import * as path from 'path';
import * as readline from 'readline';
import {Readable} from 'stream';

export class AIMockDS {
  protected _data: string[] = [];

  get data(): string[] {
    return this._data;
  }

  static async load(filePath?: string, separator = '\n'): Promise<AIMockDS> {
    const ds = new AIMockDS();
    await ds.load(filePath, separator);
    return ds;
  }

  clear() {
    this._data = [];
  }

  async load(filePath?: string, separator = '\n'): Promise<void> {
    return new Promise<void>((resolve, reject) => {
      const rootDir = path.resolve(__dirname, '../..');
      filePath = filePath ?? '';

      if (filePath === '') {
        this._data = ['This is a random response 1.', 'This is a random response 2.', 'This is a random response 3.'];
        resolve();
      } else {
        const fPath = path.isAbsolute(filePath) ? filePath : path.join(rootDir, filePath);
        if (!fs.existsSync(fPath)) {
          resolve();
          return;
        }
        const fileStream: Readable = fs.createReadStream(fPath);
        const rl = readline.createInterface({
          input: fileStream,
          crlfDelay: Infinity,
        });

        let currentLine = '';
        rl.on('line', (line: string) => {
          if (line.trim() === separator) {
            this._data.push(currentLine);
            currentLine = '';
          } else {
            currentLine += line + '\n\n';
          }
        });

        rl.on('close', () => {
          if (currentLine !== '') {
            this._data.push(currentLine);
          }
          resolve();
        });

        rl.on('error', (err: Error) => {
          reject(err);
        });
      }
    });
  }
}
