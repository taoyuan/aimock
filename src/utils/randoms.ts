import * as fs from 'fs';
import * as path from 'path';
import * as readline from 'readline';
import {Readable} from 'stream';

export namespace MockRandoms {
  let randomResponses: string[] = [];

  export async function load(filePath?: string, separator = '\n'): Promise<void> {
    return new Promise<void>((resolve, reject) => {
      const rootDir = path.resolve(__dirname, '../..');
      filePath = filePath ?? '';

      if (filePath === '') {
        randomResponses = [
          'This is a random response 1.',
          'This is a random response 2.',
          'This is a random response 3.',
        ];
        resolve();
      } else {
        const fPath = path.join(rootDir, filePath);
        if (!fs.existsSync(fPath)) {
          resolve();
          return;
        }
        const fileStream: Readable = fs.createReadStream(fPath);
        const rl = readline.createInterface({
          input: fileStream,
          crlfDelay: Infinity,
        });

        let currentLine = '';
        rl.on('line', (line: string) => {
          if (line.trim() === separator) {
            randomResponses.push(currentLine);
            currentLine = '';
          } else {
            currentLine += line + '\n\n';
          }
        });

        rl.on('close', () => {
          if (currentLine !== '') {
            randomResponses.push(currentLine);
          }
          resolve();
        });

        rl.on('error', (err: Error) => {
          reject(err);
        });
      }
    });
  }

  export const clear = () => (randomResponses = []);

  export const getRandomContents = () => randomResponses;
}
