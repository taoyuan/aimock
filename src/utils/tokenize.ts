function tokenize(content: string): string[] {
  // This regular expression matches spaces, punctuation, and individual Chinese characters.
  const regex = /(\s+|[.,!?;]|[\u4e00-\u9fa5])/g;

  // Use the regex to split the content into tokens
  return content.split(regex).filter(Boolean);
}

export {tokenize};
