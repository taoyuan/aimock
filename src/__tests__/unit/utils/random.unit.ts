import {MockRandoms} from '../../../utils/randoms';

describe('load function', () => {
  beforeEach(() => {
    MockRandoms.clear();
  });

  it('should load random responses if filePath is empty', async () => {
    await MockRandoms.load();

    const responses = MockRandoms.getRandomContents();
    expect(responses).toEqual([
      'This is a random response 1.',
      'This is a random response 2.',
      'This is a random response 3.',
    ]);
  });

  it('should load responses from the specified file', async () => {
    const filePath = '/data/contents.txt';

    await MockRandoms.load(filePath);

    const responses = MockRandoms.getRandomContents();
    // Add your expectations based on the contents of the file
    expect(responses).toMatchSnapshot();
  });

  it('should load empty responses for non-existent file', async () => {
    const filePath = 'non_existent_file.txt'; // A non-existent file path

    await MockRandoms.load(filePath);

    const responses = MockRandoms.getRandomContents();
    expect(responses).toEqual([]);
  });
});

describe('randomContents function', () => {
  it('should return an array of random responses', () => {
    // Assuming you have already loaded some random responses
    // using the load function before this test case
    const responses = MockRandoms.getRandomContents();

    // Add your expectations based on the loaded random responses
    expect(Array.isArray(responses)).toBe(true);
    expect(responses.length).toBeGreaterThanOrEqual(0);
  });
});
