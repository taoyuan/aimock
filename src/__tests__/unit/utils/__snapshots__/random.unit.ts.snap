// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`load function should load responses from the specified file 1`] = `
[
  "Hi there! How can I assist you today? 😊

This is a break line text.

@@@@

Here are a few suggestions based on what you've typed so far:



\`\`\`js

const suggestions = [

  "How to book a flight",

  "What are the best hotels in New York",

  "Things to do in Tokyo"

];

\`\`\`



Thank you for your message. Our team will get back to you as soon as possible. 🙏

@@@@

Here's some Python code:

This is a break line text.



\`\`\`py

def greet(name):

  print('Hello, ' + name + '!')

\`\`\`

",
]
`;
